app/domain/authentication/authn_k8s/k8s_resolver.rb

```diff
@@ -65,101 +55,116 @@ def pod_owner_refs
 
         # Validates that the +pod+ belongs to the resource object.
         #
-        # @exception ValidationError if the +pod+ does not belong to the resource object.
+        # @exception PodNameMismatchError,PodRelationMismatchError, PodMissingRelationError if the +pod+ does not
+        # belong to the resource object.
         def validate_pod
           raise "validate_pod is not implemented"
         end
       end
 
       # Tests whether the Pod's ReplicaSet belongs to the Deployment.
       class Deployment < Base
+        def replica_set_owner_refs(replica_set)
+          @replica_set_owner_refs ||= replica_set.metadata.ownerReferences
+        end
```

Comment:
You can't have a method that receives an input AND memoizes an object. If you call it with one value and then with another it won't do anything the second time.

Reply:
Also - we prefer to have methods declared after they are called for readability. I don't want to see a method implementation without knowing what it is for